langchain4j-http-client/src/main/java/dev/langchain4j/http/client/HttpClient.java

```diff
@@ -0,0 +1,32 @@
+package dev.langchain4j.http.client;
+
+import dev.langchain4j.Experimental;
+import dev.langchain4j.exception.HttpException;
+import dev.langchain4j.http.client.sse.ServerSentEventListener;
+import dev.langchain4j.http.client.sse.ServerSentEventParser;
+
+@Experimental
+public interface HttpClient {
+
+    /**
+     * TODO
+     *
+     * @param request
+     * @return
+     * @throws HttpException
+     * @throws RuntimeException
+     */
+    SuccessfulHttpResponse execute(HttpRequest request) throws HttpException, RuntimeException;
```

Comment:
I believe it would be too big for this version, but at some point we need to think about async / non-blocking clients.